components/datadog/agentparams/filepermissions/permissions.go

```diff
@@ -0,0 +1,8 @@
+package filepermissions
+
+// FilePermissions interface defines all commands that can be used to set and reset file permissions.
+// These functions are used to create args.Command that will be run by pulumi.
+type FilePermissions interface {
+	SetupPermissionsCommand(path string) string
+	ResetPermissionsCommand(path string) string
+}
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Nice to have something super generic so that we can have very custom implementations for each platform 👍